apps/web/src/views/Farms/v4/components/PoolApyButton.tsx

```diff
@@ -0,0 +1,324 @@
+  }, [lpApr, cakeApr, merklApr])
+  const combinedBoostedApr = useMemo(() => {
+    return Number(lpApr ?? 0) + Number(cakeApr?.boost ?? 0) + Number(merklApr ?? 0)
+  }, [lpApr, cakeApr, merklApr])
+  const aprTooltip = useTooltip(
+    <AprTooltip
+      pool={pool}
+      hasBoost={hasBoost}
+      combinedBaseApr={combinedBaseApr}
+      combinedBoostedApr={combinedBoostedApr}
+    />,
+    {
+      trigger: 'click',
+    },
+  )
+
+  if (!lpApr) {
+    return <Skeleton height={24} width={80} style={{ borderRadius: '12px' }} />
+  }
+
+  return (
+    <>
+      <FarmWidget.FarmApyButton
+        variant="text-and-button"
+        handleClickButton={(e) => {
+          e.stopPropagation()
+          e.preventDefault()
+          roiModal.onOpen()
+        }}
+      >
+        <TooltipText decorationColor="secondary" ref={aprTooltip.targetRef}>
+          <Flex ml="4px" mr="5px" style={{ gap: 5 }}>
+            {hasBoost && (
+              <>
+                <Text bold color="success" fontSize={16}>
+                  <>
+                    <Text bold color="success" fontSize={16} display="inline-block" mr="3px">
+                      🌿
+                      {t('Up to')}
+                    </Text>
+                    {`${displayApr(combinedBoostedApr)}%`}
+                  </>
+                </Text>
+              </>
+            )}
+            <Text style={{ textDecoration: hasBoost ? 'line-through' : 'none' }}>{displayApr(combinedBaseApr)}%</Text>
+          </Flex>
+        </TooltipText>
+      </FarmWidget.FarmApyButton>
+      {aprTooltip.tooltipVisible && aprTooltip.tooltip}
+      {pool?.protocol === 'v3' ? <V3ApyRoiModal poolInfo={pool} cakeApr={cakeApr} modal={roiModal} /> : null}
+      {['v2', 'stable'].includes(pool.protocol) ? (
+        <V2ApyRoiModal poolInfo={pool} cakeApr={cakeApr} modal={roiModal} />
+      ) : null}
+    </>
+  )
+}
+
+const V3ApyRoiModal: React.FC<{
+  poolInfo?: PoolInfo
+  cakeApr?: CakeApr[keyof CakeApr]
+  modal: UseModalV2Props
+}> = ({ poolInfo, cakeApr, modal }) => {
+  const { t } = useTranslation()
+  const existingPosition = undefined
+  const [priceTimeWindow, setPriceTimeWindow] = useState(PairDataTimeWindowEnum.DAY)
+  const { data: token0PriceUsd } = useCurrencyUsdPrice(poolInfo?.token0, { enabled: Boolean(poolInfo?.token0) })
+  const { data: token1PriceUsd } = useCurrencyUsdPrice(poolInfo?.token1, { enabled: Boolean(poolInfo?.token1) })
+
+  const cakePrice = useCakePrice()
+  const formState = useV3FormState()
+  const lmPoolLiquidity = useLmPoolLiquidity(poolInfo?.lpAddress, poolInfo?.chainId)
+
+  const prices = usePairTokensPrice(poolInfo?.lpAddress, priceTimeWindow, poolInfo?.chainId, modal.isOpen)
+  const { ticks: ticksData } = useAllV3Ticks(poolInfo?.token0, poolInfo?.token1, poolInfo?.feeTier, modal.isOpen)
+  const { pool, ticks, price, pricesAtTicks, currencyBalances, outOfRange } = useV3DerivedInfo(
```

Comment:
some unused value: ticks, outOfRange

Reply:
just keep it here, will use when add user position